sherpa/plot/pylab_backend.py

```diff
@@ -409,12 +395,6 @@ def plot(self, x, y, *,
         except (AttributeError, KeyError):
             lw = 1.0
 
-        if xaxis:
```

Comment:
Removing these axhline calls means we should also remove the `try/except` block above as it's now unused.

I do note that I still see the issue that is mentioned in the comment - e.g.

```
>>> axes = plt.gca()
>>> axes.spines['left'].get_linewidth()
0.8
```

so using a default of 1 (which I believe it is now done) leads to a line slightly-too-wide to match the y=0 tick (if one exists).

Reply:
This is one of those cases that are really hard to do backend independent and simple. That's why this was special-cased (as in "have a separate keyword argument that is only used in one specific plot type"). I can set the default for hline/vline to 0.8, but then it's narrower for all cases.
My feeling (which is what I implemented here) is that it's not worth to have that extra complexity for all backends, but to make that line 20% thinner. We *could* assume that matplotlib is the default plotting and put the linewidth in the plots that use to used `xaxis=True` to default to 0.8 - but then we sould have that for all backends, independent of how their axis usually look like.

I think you did most of the previous plotting work - let me know if you
- can live with slightly thicker lines (as it's now)
- prefer a fixed line width of 0.8 for specific plots for all backends (even if the axis are for some reason, e.g. a different .matplotlib.rc setting different) 
- or want me to fully restore the previous behavior.

I'm for the first, because it's simpler and backend independent.